src/deploy/hosting/prepare.ts

```diff
@@ -23,6 +21,41 @@ export async function prepare(context: Context, options: HostingOptions & Option
 
     options.config.set("hosting.public", options.public);
   }
+}
+
+/**
+ * If there is a rewrite to a tagged function, add it to the deploy target.
+ */
+export async function addTaggedFunctionsToOnlyString(
+  context: Context,
+  options: HostingOptions & Options
+): Promise<boolean> {
+  // This must be called before modifying hosting config because we turn it from
+  // a scalar to an array now
+  handlePublicDirectoryFlag(options);
+
+  let addedFunctions = false;
+  for (const c of config.hostingConfig(options)) {
+    for (const r of c.rewrites || []) {
+      if (!("function" in r) || typeof r.function !== "object" || !r.function.pinTag) {
```

Comment:
add same logic for r.run.pinTag too

Reply:
Don't need it because we don't support deploying a Run service.

Reply:
oh fair point